src/Spreadsheet/ArcAssay.fs

```diff
@@ -76,13 +76,70 @@ module ArcAssay =
         with 
         | err -> failwithf "Failed while parsing metadatasheet: %s" err.Message
 
-    let isMetadataSheetName (name:string) =
+    let toMetadataCollection (assay: ArcAssay) =
+        let toRows (assay: ArcAssay) =
+            seq {          
+                yield  SparseRow.fromValues [assaysLabel]
+                yield! Assays.toRows (Some assaysPrefix) [assay]
+
+                yield SparseRow.fromValues [contactsLabel]
+                yield! Contacts.toRows (Some contactsPrefix) (List.ofSeq assay.Performers)
+            }
+        assay
+        |> toRows
+        |> Seq.map (fun row -> SparseRow.getAllValues row)
+
+    let fromMetadataCollection (collection: seq<seq<string option>>) : ArcAssay =
+        try
+            let fromRows (usePrefixes: bool) (rows: seq<SparseRow>) =
```

Comment:
Pull this (fromRows) function out to be used both in `fromMetadataSheet` and your function.

Reply:
fixed